examples/ai-ml/ray/README.md

```diff
@@ -0,0 +1,318 @@
+# Ray with Amazon EKS
+
+> 🛑 This blueprint should be considered as experimental and should only be used for proof of concept.
+
+
+This example deploys an EKS Cluster running the Ray Operator.
+
+- Creates a new sample VPC, 3 Private Subnets and 3 Public Subnets
+- Creates Internet gateway for Public Subnets and NAT Gateway for Private Subnets
+- Creates EKS Cluster Control plane with public endpoint (for demo reasons only) with one managed node group
+- Deploys Ray Operator, AWS Load Balancer Controller, Ingress-nginx and External DNS (optional) add-ons
+- Deploys a Ray Cluster in the `ray-cluster` namespace
+
+## Prerequisites
+
+Ensure that you have installed the following tools on your machine.
+
+1. [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+4. [python3](https://www.python.org/)
+5. [jq](https://stedolan.github.io/jq/)
+
+Additionally for end-to-end configuration of Ingress, you can optionally provide the following:
+
+1. A [Route53 Public Hosted Zone](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/dns-configuring.html) configured in the account where you are deploying this example. E.g. "bar.com"
+2. An [ACM Certificate](https://docs.aws.amazon.com/acm/latest/userguide/gs-acm-request-public.html) in the account + region where you are deploying this example. A wildcard certificate is preferred, e.g. "*.bar.com"
+
+## Deploy the EKS Cluster with Ray Operator
+
+### Clone the repository
+
+```sh
+git clone https://github.com/aws-ia/terraform-aws-eks-blueprints.git
+```
+
+### Build Docker Image for Ray Cluster
+
+First, in order to run the RayCluster, we need to push a container image to ECR repository that contains the all the dependencies. You can see the [Dockerfile](sample-job/Dockerfile) to see the python dependencies packaged in. The next series of steps we will setup an ECR repository, build the docker image for our model and push it to the ECR repository.
+
+Create an ECR repository
+
+```sh
+aws ecr create-repository --repository-name ray-demo
+```
+
+Login to the ECR repository
+
+```sh
+aws ecr get-login-password \
+  --region $AWS_REGION | docker login \
+  --username AWS \
+  --password-stdin $ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/ray-demo
+```
+
+Build the docker image containing our model deployment.
+
+```sh
+docker build sources -t $ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/ray-demo
+```
+
+Push the docker image to the ECR repo
+
+```sh
+docker push $ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/ray-demo
+```
+
+### Initialize Terraform
+
+Navigate into the example directory and run `terraform init`
+
+```sh
+cd examples/ai-ml/ray/
+terraform init
+```
+
+### Terraform Plan
+
+Run Terraform plan to verify the resources created by this execution.
+
+```sh
+export AWS_REGION=<enter-your-region>
```

Comment:
It looks like these should move up to be earlier in the steps (used in the ECR steps so move to first step)